plugins/offline-storage.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Offline Storage plugin
+ * 
+ * Stores errors failed to get send and try to send them when
+ * Networkf comes back or on init
+ */
+
+var offlineStorageKey = 'raven-js-offline-queue';
+
+function offlineStoragelugin(Raven, options) {
+  options = options || {};
+
+  function processOfflineQueue() {
+    // Let's stop here if there's no connection
+    if (!navigator.onLine) {
+      return;
+    }
+
+    try {
+      // Get the queue
+      var queue = JSON.parse(localStorage.getItem(offlineStorageKey)) || [];
+
+      // Store an empty queue. If processing these one fails they get back to the queue
+      localStorage.setItem(offlineStorageKey, JSON.stringify([]));
+
+      queue.forEach(function processOfflinePayload(data) {
+        // Avoid duplication verification for offline stored
+        // as they may try multiple times to be processed
+        Raven._lastData = null;
```

Comment:
I'm not sure if I understand your comment here.
There shouldn't be any duplicates here already, as `failure` event is triggered at the end of `_sendProcessedPayload` which already checks for duplicates first.

Reply:
Yes, but then when trying to send it again when get back offline, because `_lastData` is set with potentially the same data we will be try to send, it will be missed because it's the same data.

Reply:
Right :)